### PR TITLE
replace contact by webid's into group members list

### PR DESCRIPTION
**Pull request by bourgeoa**

### CODE REVIEW

File: groupMembershipControl.js [hunk 1]
```diff
@@ -15,7 +15,18 @@ export async function renderGroupMemberships (person, context) {
   async function removeFromGroup (thing, group) {
     const pname = kb.any(thing, ns.vcard('fn'))
     const gname = kb.any(group, ns.vcard('fn'))
-    const groups = kb.each(null, ns.vcard('hasMember'), thing)
+    // find all webids of thing
+    const thingwebids = kb.each(null, ns.owl('sameAs'), thing, group.doc())
+    // webid can be deleted only if not used in a other thing
```
Comment (TallTed) [inline groupMembershipControl.js:20]:
```suggestion
    // find all WebIDs of thing
    const thingwebids = kb.each(null, ns.owl('sameAs'), thing, group.doc())
    // WebID can be deleted only if not used in another thing
```

File: groupMembershipControl.js [hunk 2]
```diff
@@ -52,9 +64,11 @@ export async function renderGroupMemberships (person, context) {
     return tr
   }
 
+  // find all documents where person ns.vcard('fn')
   function syncGroupList () {
-    const groups = kb.each(null, ns.vcard('hasMember'), person)
-
+    // to be changed person and or webids
```
Comment (TallTed) [inline groupMembershipControl.js:69]:
```suggestion
    // person and/or WebIDs to be changed
```

File: webidControl.js [hunk 3]
```diff
@@ -51,12 +51,18 @@ export async function addWebIDToContacts (person, webid, urlType, kb) {
     $rdf.st(vcardURLThing, ns.rdf('type'), urlType, person.doc()),
     $rdf.st(vcardURLThing, ns.vcard('value'), webid, person.doc())
   ]
+  // insert webID in groups
+  // replace person with webId in vcard:hasMember (webId may already exist)
```
Comment (TallTed) [inline webidControl.js:55]:
```suggestion
  // insert WebID in groups
  // replace person with WebID in vcard:hasMember (WebID may already exist)
```

File: webidControl.js [hunk 4]
```diff
@@ -51,12 +51,18 @@ export async function addWebIDToContacts (person, webid, urlType, kb) {
...
+  // insert owl:sameAs
   const groups = kb.each(null, ns.vcard('hasMember'), person)
+  const deletables = []
   groups.forEach(group => {
+    deletables.push($rdf.st(group, ns.vcard('hasMember'), person, group.doc()))
+    insertables.push($rdf.st(group, ns.vcard('hasMember'), kb.sym(webid), group.doc())) // may exist do we need to check ?
```
Comment (TallTed) [inline webidControl.js:61]:
```suggestion
    insertables.push($rdf.st(group, ns.vcard('hasMember'), kb.sym(webid), group.doc())) // May exist; do we need to check?
```

**Review by timbl — APPROVED** [review]
LGTM though not traced through all the logic

**Pull request merged**